### PR TITLE
UHM-6747, navigate to the invalid fields within the current section

### PR DESCRIPTION
**Pull request by cioan**

### CODE REVIEW

File: omod/src/test/webapp/resources/scripts/navigatorHandlers.js [hunk 1]
```diff
@@ -541,15 +541,21 @@ describe("Tests for simple form navigation handlers", function() {
...
                 firstSection.isSelected = true;
                 firstSection.questions = [question];
 
                 clickedSectionHandler(sections, secondSection, event);
 
-                expect(firstSection.onExit).not.toHaveBeenCalled();
+                expect(firstSection.onExit).toHaveBeenCalled();
                 expect(event.stopPropagation).toHaveBeenCalled();
-                expect(field.select).toHaveBeenCalled();
+                expect(field.select).not.toHaveBeenCalled();
```
Comment (mogoodrich) [inline omod/src/test/webapp/resources/scripts/navigatorHandlers.js:558]: Strange, changing this test seems incorrect... ?

Basically, we are setting the first section to have it's validation fail on line 542... and the behavior is that exit handlers should be called after validation handlers, so if the validation fails, it should stay in the first section and the exit handle should have been called.

And I think this is then specifically testing that it jumps to the first question in the section if the section is invalid, so "field.select" should be true?

Thinking it through, you likely need to have the new spyOn you added return values that mock the error state.... we want to mock a situation where there is an error on the field within the first section, so spyOn(field, "isValid")should return false, and spyOn(firstSection, "firstInvalidQuestion") should return "question", and spyOn(question, firstInvalidField) should return "field", I think?

**Review by mogoodrich — COMMENTED** [review]
Hey @cioan , change looks good, but I had some comments on the test you updated, thanks!

File: omod/src/test/webapp/resources/scripts/navigatorHandlers.js (hunk 1 above)
Comment (cioan) [reply omod/src/test/webapp/resources/scripts/navigatorHandlers.js:559]: Thanks @mogoodrich ! I have updated the spyOn functions you mentioned. But it does not have an impact on the outcome of the test. I think that now, even if in the UI is not existing the section, behind the scenes it exists and it enters again to the same section and to the invalid field? Maybe this entire test case should be re-written?

Comment (mogoodrich) [reply omod/src/test/webapp/resources/scripts/navigatorHandlers.js:559]: Sure, feel free to try rewriting the test case @cioan .  don't worry about getting too bogged down, I'm happy to take a stab at it when I'm back from vacation.

**Pull request merged**